lib/diversity/component.rb

```diff
@@ -56,30 +56,26 @@ def initialize(spec, options)
       @options = DEFAULT_OPTIONS.keep_merge(options)
       @logger = @options[:logger]
 
-      schema = JsonSchemaCache[
-                 MASTER_COMPONENT_SCHEMA,
-                 { validate_spec: @options[:validate_spec] }
-               ]
-
       # Handle both Hash and String specs
       @raw = (spec.is_a?(String) ? parse_config(spec) : spec)
 
-      validation = schema.validate(@raw)
-      raise Diversity::Exception, "Configuration is not valid:\n" + validation.join("\n"), caller if
-        validation.length > 0
+      if (@options[:validate_spec])
+        schema = JsonSchemaCache[
+          MASTER_COMPONENT_SCHEMA,
+          { validate_spec: @options[:validate_spec] }
+        ]
+
+        validation = schema.validate(@raw)
+        raise Diversity::Exception,
+              "Configuration is not valid:\n" + validation.join("\n"), caller if
+          validation.length > 0
+      end
 
-      @checksum = Digest::SHA1.hexdigest(dump)
+      @checksum = "#{name}:#{version}"
```

Comment:
We need to simplify a lot of extrachecks.  A cachegrind showed that about 25% of execution goes to handle Gem::Version, and about the same went to ComponentSet << due to the equality checks.  Not as much to initiation though, so the main performance increase here was removing is_a :P